docs/spec/v2alpha1/helmreleases.md

```diff
@@ -0,0 +1,276 @@
+# Helm Releases
+
+// TODO(hidde): short conclusive API description / outline
+
+## Specification
+
+A **helmrelease** object defines the source of a Helm chart by referencing an object
+managed by [source-controller](https://github.com/fluxcd/source-controller), the interval
+reconciliation should happen at, and a set of options to control the settings of the
+automated Helm actions that are being performed.
+
+```go
+type HelmReleaseSpec struct {
+	// ReleaseName used for the Helm release. Defaults to a composition of
+	// '[TargetNamespace-]Name'.
+	// +optional
+	ReleaseName string `json:"releaseName,omitempty"`
+
+	// TargetNamespace to target when performing operations for the HelmRelease.
+	// Defaults to the namespace of the HelmRelease.
+	// +optional
+	TargetNamespace string `json:"targetNamespace,omitempty"`
+
+	// SourceRef of the Helm chart source.
+	// +required
+	SourceRef corev1.TypedLocalObjectReference `json:"sourceRef"`
+
+	// DependsOn holds a list of HelmReleases that must be ready before this
+	// release can be installed.
+	// +optional
+	DependsOn []string `json:"dependsOn,omitempty"`
```

Comment:
Noticed we're using `string` here to refer to `HelmRelease`s, while using `corev1.TypedLocalObjectReference` above to reference `HelmChart`s.  Why this difference?

Reply:
Because a `TypedLocalObjectReference` can refer to other types than `HelmRelease` resources, and you should only be able to depend on things within the same namespace, which makes this a simple `metadata.Name` list.